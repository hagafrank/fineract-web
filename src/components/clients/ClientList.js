import React from "react";
import PropTypes from "prop-types";
import { FormattedMessage } from "react-intl";

const ClientList = ({ clients }) => (
  <table className="table">
    <thead>
      <tr>
        <th>
          <FormattedMessage id="ClientList.name" />
        </th>
        <th>
          <FormattedMessage id="ClientList.clientNumber" />
        </th>
        <th>
          <FormattedMessage id="ClientList.externalId" />
        </th>
        <th>
          <FormattedMessage id="ClientList.status" />
        </th>
        <th>
          <FormattedMessage id="ClientList.office" />
        </th>
        <th>
          <FormattedMessage id="ClientList.staff" />
        </th>
      </tr>
    </thead>
    <tbody>
      {clients.map((client) => {
        return (
          <tr key={client.id}>
            <td>{client.displayName}</td>
            <td>{client.accountNo}</td>
            <td>{client.externalId}</td>
            <td>{client.status.value}</td>
            <td>{client.officeName}</td>
            <td>{client.staffName}</td>
          </tr>
        );
      })}
    </tbody>
  </table>
);

ClientList.propTypes = {
  clients: PropTypes.array.isRequired
};

export default ClientList;
