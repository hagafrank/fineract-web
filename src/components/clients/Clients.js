import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import ClientList from "./ClientList";
import Spinner from "../common/Spinner";
import { FormattedMessage } from "react-intl";
import { loadClients } from "../../redux/actions/clientActions";

class Clients extends React.Component {
  constructor() {
    super();

    this.state = {
      loading: false
    };
  }

  toggleSpinner = (show) => {
    this.setState({ loading: show });
  };

  componentDidMount() {
    this.toggleSpinner(true);
    this.props
      .loadClients()
      .then(() => this.toggleSpinner(false))
      .catch((error) => {
        this.toggleSpinner(false);
        console.error("Loading clients failed. " + error);
      });
  }

  render() {
    return (
      <>
        <h2>
          <FormattedMessage id="Clients.header" />
        </h2>
        {this.state.loading ? (
          <Spinner />
        ) : (
          <ClientList clients={this.props.clients} />
        )}
      </>
    );
  }
}

Clients.propTypes = {
  clients: PropTypes.array.isRequired,
  loadClients: PropTypes.func.isRequired
};

function mapStateToProps(state) {
  return {
    clients: state.clients.clients || []
  };
}

const mapDispatchToProps = {
  loadClients
};

export default connect(mapStateToProps, mapDispatchToProps)(Clients);
