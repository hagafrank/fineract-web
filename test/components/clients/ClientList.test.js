import React from "react";
import { clients } from "../../mock_data/clients/clients";
import ClientList from "../../../src/components/clients/ClientList";
import createComponentWithIntl from "../../helpers/jest-snapshot/createComponentWithIntl";

test("clients list should be rendered", () => {
  const component = createComponentWithIntl(<ClientList clients={clients} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
