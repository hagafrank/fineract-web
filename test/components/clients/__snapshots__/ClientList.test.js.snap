// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clients list should be rendered 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Client #
      </th>
      <th>
        External Id
      </th>
      <th>
        Status
      </th>
      <th>
        Office
      </th>
      <th>
        Staff
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        IBANERA
      </td>
      <td>
        excutive c000000123
      </td>
      <td>
        243567896787
      </td>
      <td>
        Active
      </td>
      <td>
        Head Office
      </td>
      <td />
    </tr>
    <tr>
      <td>
        Robert Jakech
      </td>
      <td>
        000000001
      </td>
      <td />
      <td>
        Active
      </td>
      <td>
        Head Office
      </td>
      <td />
    </tr>
    <tr>
      <td>
        Moses Kale√±a
      </td>
      <td>
        000000002
      </td>
      <td />
      <td>
        Active
      </td>
      <td>
        Head Office
      </td>
      <td />
    </tr>
    <tr>
      <td>
        Luis Espinoza
      </td>
      <td>
        000000003
      </td>
      <td />
      <td>
        Active
      </td>
      <td>
        Head Office
      </td>
      <td />
    </tr>
    <tr>
      <td>
        prueba
      </td>
      <td>
        000000004
      </td>
      <td />
      <td>
        Active
      </td>
      <td>
        Head Office
      </td>
      <td />
    </tr>
  </tbody>
</table>
`;
